### PR TITLE
[Variation] Component docs update

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?     | no
| Fixed tickets |

### CODE REVIEW

File: components/Variation/basic_usage.rst [hunk 1]
```diff
@@ -36,21 +41,18 @@ These variants should be generated by the ``VariantGenerator`` from the options
...
 .. code-block:: php
 
     use Sylius\Component\Resource\Repository\RepositoryInterface;
     use Sylius\Component\Variation\Generator\VariantGenerator;
     use Sylius\Component\Variation\Model\VariableInterface;
     use Sylius\Component\Variation\SetBuilder\CartesianSetBuilder;
- 
+
     /** @var RepositoryInterface $variantRepository */
     $variantRepository = /*...*/
```
Comment (tuka217) [inline components/Variation/basic_usage.rst]: Maybe it should be in memory repository?

File: components/Variation/basic_usage.rst [hunk 2]
```diff
@@ -36,21 +41,18 @@ These variants should be generated by the ``VariantGenerator`` from the options
...
 
     $setBuilder = new CartesianSetBuilder();
-    
+
     /**
      * From the example above, this would be the T-Shirt product.
      *
```
Comment (tuka217) [inline components/Variation/basic_usage.rst]: Istead of using comment maybe create an instance of class which implements VariableInterface.

File: components/Variation/interfaces.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+In order for the object class to manage variants and options it has to implement the ``VariableInterface``.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API VariableInterface`_.
+
+.. _Sylius API VariableInterface: http://api.sylius.org/Sylius/Component/Variation/Model/VariableInterface.html
+
+
```
Comment (tuka217) [inline components/Variation/interfaces.rst]: Additional blank line

File: components/Variation/interfaces.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+~~~~~~~~~~~~~~~~~
+
+In order for the object class to manage variants and options it has to implement the ``VariableInterface``.
+
+.. note::
+
+    You will find more information about this interface in `Sylius API VariableInterface`_.
+
+.. _Sylius API VariableInterface: http://api.sylius.org/Sylius/Component/Variation/Model/VariableInterface.html
+
```
Comment (tuka217) [inline components/Variation/interfaces.rst]: Additional blank space

File: components/Variation/models.rst [hunk 5]
```diff
@@ -1,132 +1,79 @@
...
-| setDefaults(VariantInterface $masterVariant) | This method is used toinherit values from a master variant          | Void                       |
-+----------------------------------------------+---------------------------------------------------------------------+----------------------------+
+Every variant is represented by **Variant** instance and has the following properties:
+
++--------------+---------------------------------------------+
+| Method       | Description                                 |
++==============+=============================================+
+| master       | Defines whether variant is master           |
++--------------+---------------------------------------------+
+| presentation | Name displayed to user.                     |
```
Comment (tuka217) [inline components/Variation/models.rst]: Dot :)

File: components/Variation/models.rst [hunk 6]
```diff
@@ -1,132 +1,79 @@
...
-+---------------------------------------------+------------------------------------------+----------------------------+
\ No newline at end of file
+Every variant option value is represented by **OptionValue** instance and has the following properties:
+
++--------------+---------------------------------------------+
+| Method       | Description                                 |
++==============+=============================================+
+| value        | Option internal value                       |
++--------------+---------------------------------------------+
+| option       | Option                                      |
```
Comment (NeverResponse) [inline components/Variation/models.rst]: Please provide a little more specific description.

File: components/Variation/models.rst [hunk 7]
```diff
@@ -1,132 +1,79 @@
...
++--------------+---------------------------------------------+
+| createdAt    | Date of creation                            |
++--------------+---------------------------------------------+
+| updatedAt    | Date of the last update                     |
++--------------+---------------------------------------------+
+
+.. note::
+
+    This model implements the :ref:`component_variation_model_variant-interface`.
+
```
Comment (NeverResponse) [inline components/Variation/models.rst]: Missing link to Sylius API. Same goes to other models.

File: components/Variation/basic_usage.rst [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+         */
+        public function hasOption(OptionInterface $option)
+        {
+            if ($this->options->contains($option)) {
+                return true;
+            }
+
+            return false;
+        }
+    }
```
Comment (pjedrzejewski) [inline components/Variation/basic_usage.rst]: Maybe we could consider skipping some of the methods, if their implementation does not include anything important to show ... OK, I think we should keep them and refactor. :>

**Comment by pjedrzejewski** [comment]
Thank you @TheMadeleine! :+1:

**Pull request merged**